bigquery/google/cloud/bigquery/client.py

```diff
@@ -256,14 +266,18 @@ def list_projects(self, max_results=None, page_token=None, retry=DEFAULT_RETRY):
 
             retry (google.api_core.retry.Retry): (Optional) How to retry the RPC.
 
+            timeout (Optional[float]):
+                The number of seconds to wait for the underlying HTTP transport
+                before using ``retry``.
```

Comment:
Is a timeout considered retryable by our default retry config? https://github.com/googleapis/google-cloud-python/blob/master/bigquery/google/cloud/bigquery/retry.py

If we don't have one yet, let's test that the request is retried after a timeout.

Reply:
It appears not, but I can double check later.

Reply:
**Update:** `retry` occurs in a few tests, but it's mostly just for checking if it is passed to a sub-method.

I will add an explicit test for a retry after a timeout for the `Client._call_api()` method, as it is a core helper method used at a lot of places.